### ISSUE TITLE
Rex::Pkg::Redhat is_installed() dosen't check the version correctly

### ISSUE THREAD

**Issue by klesteb** [RexOps/Rex#649]
RHEL uses the epoch, version and release to define the "version" of a rpm. So a version would be
0:0.01-1.el6, where the epoch is optional.

is_installed() only checks the version, thereby never finding the correct package. 

A possible fix would be to add a "query" method, such as Rex::Pkg::FreeBSD to find the correct package.

**Comment by krimdomu** [comment]
Hi,
thanks for the report. We're going to look at it.

**Comment by klesteb** [comment]
It might be a mote point, but the is_installed() method in Rex::Pkg doesn't pass on any options to specific is_installed().

**Comment by klesteb** [comment]
Sorry, that's Rex::Command::Pkg.

**Comment by ferki** [comment]
@klesteb: yes, thanks for pointing that out! You're right about that the `is_installed` command (`Rex::Commands::is_installed()`) does not pass any other argument than the package name to the underlying OS-specific implementation, so that could be part of the fix (e.g. optionally accept a version parameter and if present, also try to check for the exact version).

On the other hand we need to debug if there's something wrong with gathering detailed Red Hat package information.

**Comment by klesteb** [comment]
Understood. It looks like you are doing it correctly, just that version means something differently for rpms.

Same problem in Rex::Pkg::Base for diff_package_list().
